plugin/src/shared/generators/navigation-utils/next-app-lib-files/shared/utils/navigation/src/lib/utils/does-route-match.ts.template

```diff
@@ -0,0 +1,13 @@
+import { isInteger } from 'lodash-es';
+import type { Route } from '../types/route.ts';
+
+export const doesRouteMatch = (pathname: string, route: Route): boolean => {
+  if (typeof route === 'function') {
+    const segments = pathname.split('/');
+    const numericSegment = segments.find((segment) => isInteger(parseInt(segment)));
```

Comment:
@ekazankova What if we'll have more than one numeric segment in the route? Shouldn't we support this case as well?